ios/brave-ios/Sources/Brave/Frontend/Browser/New Tab Page/NewTabPageViewController.swift

```diff
@@ -487,20 +497,22 @@ class NewTabPageViewController: UIViewController {
     )
   }
 
-  func setupBackgroundVideoIfNeeded() {
+  func setupBackgroundVideoIfNeeded(shouldCreatePlayer: Bool) {
     videoButtonsView.isHidden = true
 
-    // Setup the background video only if:
-    // - it hasn't been setup before
-    // - the current NTP background is a video
-    guard videoPlayer == nil,
-      let backgroundVideoPath = background.backgroundVideoPath
-    else {
+    guard let backgroundVideoPath = background.backgroundVideoPath else {
+      videoPlayer = nil
+      backgroundView.resetPlayerLayer()
```

Comment:
We need to free resources and hide play button when the current NTP is changed from NTT video to image background.